siliconcompiler/core.py

```diff
@@ -4308,7 +4327,7 @@ def run(self):
                 index_succeeded = False
                 for index in indexlist[step]:
                     stepstr = step + index
-                    if status[stepstr] != TaskStatus.ERROR:
+                    if stepstr in status and status[stepstr] != TaskStatus.ERROR:
```

Comment:
What prompted this extra check? Seems like this could only be False if a step in the steplist or index in the indexlist is not part of the flowgraph, which would be problematic.

Reply:
It's probably safe to remove this. I briefly tried adding a 'screenshot' task to the steplist on the server side, which raised some issues around here, but that was a dead end.